spec/lib/query/query_event_metrics_spec.rb

```diff
@@ -0,0 +1,112 @@
+
+  let!(:nonprofit2) { create(:nonprofit, slug: SecureRandom.uuid) }
+  let!(:other_event_active2) { create(:event, nonprofit: nonprofit2, end_datetime: 10.days.since, published: true, slug: SecureRandom.uuid) }
+  let!(:other_event_active3) { create(:event, nonprofit: nonprofit2, end_datetime: 20.days.since, published: true, slug: SecureRandom.uuid) }
+
+  describe ".for_listings" do
+    describe "for nonprofit" do
+      describe "active" do
+        subject(:results) { QueryEventMetrics.for_listings("nonprofit", nonprofit.id, {"active" => true}) }
+
+        it "query result has correct attributes" do
+          expect(results.count).to eq 2
+
+          # first element is event_active2 because of order desc
+          expect(results.first["id"]).to eq event_active2.id
+          expect(results.first["name"]).to eq event_active2.name
+          expect(results.first["venue_name"]).to eq event_active2.venue_name
+          expect(results.first["address"]).to eq event_active2.address
+          expect(results.first["city"]).to eq event_active2.city
+          expect(results.first["state_code"]).to eq event_active2.state_code
+          expect(results.first["zip_code"]).to eq event_active2.zip_code
+          expect(results.first["start_datetime"]).to eq 10.days.ago.at_beginning_of_day
+          expect(results.first["end_datetime"]).to eq 20.days.since.at_beginning_of_day
+          expect(results.first["organizer_email"]).to eq event_active2.organizer_email
+          expect(results.first["checked_in_count"]).to eq 1
+          expect(results.first["total_attendees"]).to eq 7
+          expect(results.first["tickets_total_paid"]).to eq 3_222
+          expect(results.first["donations_total_paid"]).to eq 7_959
+          expect(results.first["total_paid"]).to eq(3_222 + 7_959)
+
+          expect(results.last["id"]).to eq event_active1.id
+        end
+      end
+
+      describe "active" do
+        subject(:results) { QueryEventMetrics.for_listings("nonprofit", nonprofit.id, {"past" => true}) }
+        it "query result has correct element" do
+          expect(results.count).to eq 1
+          expect(results.first["id"]).to eq event_past.id
+        end
+      end
+
+      describe "unpublished" do
+        subject(:results) { QueryEventMetrics.for_listings("nonprofit", nonprofit.id, {"unpublished" => true}) }
+        it "query result has correct element" do
+          expect(results.count).to eq 1
+          expect(results.first["id"]).to eq event_unpublished.id
+        end
+      end
+
+      describe "deleted" do
+        subject(:results) { QueryEventMetrics.for_listings("nonprofit", nonprofit.id, {"deleted" => true}) }
+        it "query result has correct element" do
+          expect(results.count).to eq 1
+          expect(results.first["id"]).to eq event_deleted.id
+        end
+      end
+    end
+
+    describe "for profile" do
+      skip "TODO"
+    end
```

Comment:
Not an issue but a fun little feature of rspec I learned just in the last week.

If you write just `describe "for profile"` or `it` without a block, rspec treats it as if you told it to skip.